lib/protocol/message.go

```diff
@@ -126,14 +127,24 @@ func (o *ClusterConfigMessage) GetOption(key string) string {
 }
 
 type Folder struct {
-	ID      string   // max:256
-	Devices []Device // max:1000000
-	Flags   uint32
-	Options []Option // max:64
+	ID              string // max:256
+	ReadOnly        bool
+	RescanIntervalS int
+	IgnorePerms     bool
+	AutoNormalize   bool
+	Order           int
+	IgnoreDelete    bool
+	Devices         []Device // max:1000000
+	Flags           uint32
+	Options         []Option // max:64
 }
 
 type Device struct {
-	ID              []byte // max:32
+	ID              []byte   // max:32
+	Name            string   // max:64
+	Addresses       []string // max:64
```

Comment:
This needs a struct type, or we can't limit the length of the strings. (Or you could improve the xdr generator to handle this ;)